index.js

```diff
@@ -422,6 +424,20 @@ Sharp.prototype.overlayWith = function(overlay, options) {
 };
 
 /*
+  Add another color channel to the image
+*/
+Sharp.prototype.joinChannel = function(images, options) {
+  if (Array.isArray(images)) {
+    images.forEach(function(image, index) {
+      this.options.joinChannelIn[index] = this._createInputDescriptor(image, options);
```

Comment:
Should this use `push()` rather than be index based? This would allow `.joinChannel(array1).joinChannel(array2)`.